### PR TITLE
Axe cli update

### PR DESCRIPTION
**Pull request by WilcoFiers**
Update that will enable running axe-cli (soon to be released).

Changes include:
- More tests
- Contributor.md doc
- Bugfix to running rules with certain tags
- Allow passing axe-core source as a parameter (optional)

### CODE REVIEW

**Comment by marcysutton** [comment]
@WilcoFiers what are all of these other commits?

**Comment by marcysutton** [comment]
@WilcoFiers @dylanb there are a ton of old commits, I assume from Bitbucket, included in this PR. How do we resolve the CLA for Sturley's account since he has commits in there? Perhaps we should limit this PR to only the things relevant to axe-cli instead of the entire history?

**Comment by WilcoFiers** [comment]
I don't much like that. There is a bunch of good tests in here that we should be sharing. Can we rebase to get around the issue?

**Comment by dylanb** [comment]
Ignore the CLA issues : these are all Deque employees/contractors where we have signed agreements on IP

**Comment by marcysutton** [comment]
@WilcoFiers I'm not sure where you'd start a rebase from–some of the older commits from Bitbucket don't exist in this repo, while some represent content that already exists (like Contributing.md). The older commits didn't seem critical to this PR, but untangling them would be more trouble than it's worth. The merge conflicts look pretty easy to resolve.

**Comment by dylanb** [comment]
if we don't need to worry about the CLA issues, then what reason is there left to rebase at all?

File: lib/inject.js [hunk 1]
```diff
@@ -38,11 +38,12 @@ function findFramesAndInject(parent, script, driver) {
...
  * @param {Function} callback Callback to execute when source is retrieved
  */
-function getSource(callback) {
-	if (AXE_SOURCE) {
-		return callback(null, AXE_SOURCE);
+function getSource(axeSource, callback) {
+	if (axeSource !== null) {
+		return callback(null, axeSource);
 	}
 	var pathToAxe = path.resolve(__dirname, '../node_modules/axe-core/axe.js');
```
Comment (dylanb) [inline lib/inject.js]: @WilcoFiers This diff is f##%#ed, but...why did you change the collection of the source code from a require to a file read?

File: test/unit/index.js [hunk 2]
```diff
@@ -147,7 +147,7 @@ describe('Builder', function () {
 		it('should normalize context', function (done) {
 			var normalized = false;
 			var Builder = proxyquire('../../lib/index', {
-				'./inject': function (driver, cb) {
+				'./inject': function (driver, source, cb) {
```
Comment (marcysutton) [inline test/unit/index.js]: These proxyquire tests are blowing up with Selenium Webdriver 3: 

```
Running "mochaTest:integration" (mochaTest) task
>> Mocha exploded!
>> SyntaxError: Unexpected token ...
>>   at exports.runInThisContext (vm.js:53:16)
>>   at Module._compile (module.js:387:25)
>>   at Object.Module._extensions..js (module.js:422:10)
>>   at Module.load (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
>>   at Function.Module._load (module.js:314:12)
>>   at Module.require (module.js:367:17)
>>   at require (internal/module.js:20:19)
>>   at Object.<anonymous> (/Users/marcysutton/Sites/Deque/axe-webdriverjs/test/run-webdriver.js:1:79)
>>   at Module._compile (module.js:413:34)
>>   at Object.Module._extensions..js (module.js:422:10)
>>   at Module.load (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
>>   at Function.Module._load (module.js:314:12)
>>   at Module.require (module.js:367:17)
>>   at require (internal/module.js:20:19)
>>   at Object.<anonymous> (/Users/marcysutton/Sites/Deque/axe-webdriverjs/test/integration/doc-dylang.js:6:20)
>>   at Module._compile (module.js:413:34)
>>   at Object.Module._extensions..js (module.js:422:10)
>>   at Module.load (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
>>   at Function.Module._load (module.js:314:12)
>>   at Module.require (module.js:367:17)
>>   at require (internal/module.js:20:19)
>>   at /Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/mocha/lib/mocha.js:222:27
>>   at Array.forEach (native)
>>   at Mocha.loadFiles (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/mocha/lib/mocha.js:219:14)
>>   at Mocha.run (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/mocha/lib/mocha.js:487:10)
>>   at MochaWrapper.run (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt-mocha-test/tasks/lib/MochaWrapper.js:44:13)
>>   at /Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt-mocha-test/tasks/mocha-test.js:84:20
>>   at capture (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt-mocha-test/tasks/mocha-test.js:31:5)
>>   at Object.<anonymous> (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt-mocha-test/tasks/mocha-test.js:79:5)
>>   at Object.<anonymous> (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt/lib/grunt/task.js:255:15)
>>   at Object.thisTask.fn (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt/lib/grunt/task.js:73:16)
>>   at Object.<anonymous> (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt/lib/util/task.js:294:30)
>>   at Task.runTaskFn (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt/lib/util/task.js:244:24)
>>   at Task.<anonymous> (/Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt/lib/util/task.js:293:12)
>>   at /Users/marcysutton/Sites/Deque/axe-webdriverjs/node_modules/grunt/lib/util/task.js:220:11
>>   at _combinedTickCallback (internal/process/next_tick.js:67:7)
>>   at process._tickDomainCallback (internal/process/next_tick.js:122:9)
Warning: Task "mochaTest:integration" failed. Use --force to continue.
```

**Comment by marcysutton** [comment]
I rebased on this repo to resolve the conflicts. The `configure` API method was missed somehow, so I added it back in. But the tests do not pass with Selenium Webdriver 3.

File: lib/inject.js (hunk 1 above)
Comment (marcysutton) [inline lib/inject.js]: That is most of the point of this PR–to allow the source to be passed in as a parameter, or alternatively, look for it in the node_modules.

File: test/unit/index.js (hunk 2 above)
Comment (marcysutton) [inline test/unit/index.js]: Derp. I was using an older version of Node. Using 6.9.0 works.

**Comment by marcysutton** [comment]
I still can't run the sauce integration test–but otherwise this PR works. I think we should remove that one but I'll leave it for another day.

@WilcoFiers can you address @dylanb's comment about the axe source and then I'll merge?

**Comment by WilcoFiers** [comment]
@marcysutton @dylanb I updated the way axe gets loaded in. I'm not sure why we had such complicated code. All we needed to do was this:

```javascript
axeSource = axeSource || require('axe-core').source
```
